plugins/modules/memory.ps1

```diff
@@ -48,8 +48,13 @@ try {
         $outputHash = @{}
         $output = Set-DbaMaxMemory @setMemorySplat
         $module.Result.changed = $true
-        foreach ($property in $output.PSObject.Properties.Name ) {
-            $outputHash[$property] = [String]$output.$property
+        foreach ($property in $output.PSObject.Properties ) {
+            if ($property.TypeNameOfValue -like "Microsoft*") {
+                $outputHash[$property.Name] = [System.String]$output.$($property.Name)
+            }
+            else {
+                $outputHash[$property.Name] = $output.$($property.Name)
+            }
         }
         $module.Result.data = $outputHash
```

Comment:
@LowlyDBA any reason to construct a hashtable instead of just returning the object? Ansible module will serialize it, so this should work:

```powershell
$module.Result.data = $output
```

Reply:
It was causing some "urn could not be resolved at level OleDbProviderSetting" issues. I had used this pattern before once or twice, so this issue is new. It _may_ have also been causing Segmentation Faults in my local WSL (though can't remember for sure about the latter issue).

Reply:
ahh yeah, a not uncommon issue, lots of types don't serialize well, look what happened when I tried to serialize `PSModuleInfo` objects 😩:
https://github.com/ansible-collections/community.windows/blob/main/plugins/modules/win_psmodule_info.ps1